src/MemberExtension.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Taitava\SilverstripeEmailQueue;
+
+use SilverStripe\Security\Member;
+use SilverStripe\ORM\DataExtension;
+
+/**
+ * Members cannot implement @link EmailAddressProvider
+ * so manually add the method required
+ */
+class MemberExtension extends DataExtension
+{
+    public function getEmailAddresses(): array
```

Comment:
I'll go through the code better later, but just a small thing I noticed now. PHPDoc could be used to denote that the return value is `string[]` (an array of strings), which is even more detailed in addition to the `: array`, which is also good.
```suggestion
    /**
     * @return string[] Email addresses as keys, member names as values.
     */
    public function getEmailAddresses(): array
```

It's been a while since I've last coded with PHP, so I don't know if it possible to define `string[]` directly as a return type like this: `public function getEmailAddresses(): string[]`?

Reply:
I tried adding `string[]` as the method output, but (at least for PHP7.4) this is not allowed.